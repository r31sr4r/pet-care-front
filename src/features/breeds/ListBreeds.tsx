import {
	Avatar,
	Box,
	Button,
	IconButton,
	List,
	ListItem,
	ListItemAvatar,
	ListItemText,
	styled,
	Tooltip,
} from '@mui/material';
import { useAppSelector } from '../../app/hooks';
import { selectBreeds, useGetBreedsQuery } from './breedsSlice';
import PetsIcon from '@mui/icons-material/Pets';
import OpenInNewIcon from '@mui/icons-material/OpenInNew';
import { Link, useNavigate } from 'react-router-dom';
import { BreedSelector } from './components/BreedSelector';
import React from 'react';

export const ListBreeds = () => {
	const { data, isFetching, error } = useGetBreedsQuery();
	const breeds = useAppSelector(selectBreeds);
	const [breedName, setBreedName] = React.useState('');
	const navigate = useNavigate();


	const Demo = styled('div')(({ theme }) => ({
		backgroundColor: theme.palette.background.paper,
	}));

	const handleClick = (breed: any) => {
		navigate(`/breeds/edit/${breed.id}`);
	};

	const handleBreedChange = (e: any) => {
		setBreedName(e.target.value);
	};


	const selectBreedType = (type: string) => {
		switch (type) {
			case 'DOG':
				return 'Cão';
			case 'CAT':
				return 'Gato';
			case 'BIRD':
				return 'Pássaro';
			case 'FISH':
				return 'Peixe';
			case 'RABBIT':
				return 'Coelho';
			case 'REPTILE':
				return 'Réptil';
			case 'OTHER':
				return '';
			default:
				return '';
		}
	};

	return (
		<Box maxWidth="lg" sx={{ mt: 4, mb: 4 }}>
			<Box display="flex" justifyContent="flex-end">
				<Button
					variant="contained"
					color="secondary"
					component={Link}
					to="/breeds/create"
					style={{ marginBottom: '1rem' }}
				>
					New Breed
				</Button>
			</Box>
			<Box>
				<BreedSelector breedName={breedName} petType='DOG' handleBreedChange={handleBreedChange} />
			</Box>
			<Demo>
				<List>
					{data?.data.map((breed: any) => (
						<ListItem
							key={breed.id}
							divider
							secondaryAction={
								<Tooltip
									title="Detalhar"
									placement="left-start"
								>
									<IconButton
										edge="end"
										aria-label="delete"
										onClick={() => handleClick(breed)}
									>
										<OpenInNewIcon />
									</IconButton>
								</Tooltip>
							}
						>
							<ListItemAvatar>
								<Avatar>
									<PetsIcon />
								</Avatar>
							</ListItemAvatar>
							<ListItemText
								primary={breed.name}
								secondary={`${selectBreedType(breed.type)} `}
							/>
						</ListItem>
					))}
				</List>
			</Demo>
		</Box>
	);
};
