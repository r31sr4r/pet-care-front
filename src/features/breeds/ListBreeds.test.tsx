import { renderWithProviders } from '../../utils/test/test-utils';
import { ListBreeds } from './ListBreeds';

describe('ListBreeds', () => {
    it('should render successfully', () => {
        const { asFragment } = renderWithProviders(<ListBreeds />);
        expect(asFragment()).toMatchSnapshot();
    });
});