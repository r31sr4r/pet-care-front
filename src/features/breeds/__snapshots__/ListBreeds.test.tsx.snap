// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBreeds should render successfully 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1ozyran"
  >
    <div
      class="MuiBox-root css-1bvc4cc"
    >
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        href="/breeds/create"
        style="margin-bottom: 1rem;"
        tabindex="0"
      >
        New Breed
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          id="demo-simple-select-label"
        >
          Raça
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="demo-simple-select-label demo-simple-select"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="demo-simple-select"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            name="breed"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Raça
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="css-1k9ek97"
    >
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
